pkg/scheduler/core/generic_scheduler.go

```diff
@@ -389,12 +394,21 @@ func (g *genericScheduler) getLowerPriorityNominatedPods(pod *v1.Pod, nodeName s
 
 // numFeasibleNodesToFind returns the number of feasible nodes that once found, the scheduler stops
 // its search for more feasible nodes.
-func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) int32 {
-	if numAllNodes < minFeasibleNodesToFind || g.percentageOfNodesToScore <= 0 ||
-		g.percentageOfNodesToScore >= 100 {
+func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes int32) {
```

Comment:
Now that this is a bit more complex, could you please write a unit-test for this function?

Reply:
> NOTE: I will add some test case about the percentage of nodes if #71926 update the scheduling result type.

Yes, in fact, I originally planned to wait for #71926 to merge and add unit test cases for this because it need to use `ScheduleResult` type.